gadgets/trace_dns/program.bpf.c

```diff
@@ -212,13 +226,13 @@ static __always_inline int load_addresses(struct __sk_buff *skb, int ancount,
 			__builtin_memset(&event->anaddr[index][0], 0x0, 10);
 			__builtin_memset(&event->anaddr[index][10], 0xff, 2);
 			bpf_skb_load_bytes(skb, rroffset + sizeof(struct dnsrr),
-					   &event->anaddr[index][12], rdlength);
+					   &event->anaddr[index][12], 4);
```

Comment:
```
	; bpf_skb_load_bytes(skb, rroffset + sizeof(struct dnsrr),
	658: (bf) r1 = r6                     ; R1_w=ctx(off=0,imm=0) R6=ctx(off=0,imm=0) refs=257
	659: (bf) r2 = r8                     ; R2_w=326 R8_w=326 refs=257
	660: (05) goto pc+48
	;
	709: (85) call bpf_skb_load_bytes#26
	R4 min value is negative, either use unsigned or 'var &= const'
	processed 3546 insns (limit 1000000) max_states_per_insn 4 total_states 80 peak_states 80 mark_read 64
```

The ebpf verifier is not able to infer that rdlength is positive.

But in this context, because of the 'if' condition, rdlength is a positive constant and is the size of an ipv4 address (4) or an ipv6 address (16). I'll replace it with sizeofs.